File: Racing/v2/Racing.py
import tkinter as tk
import random

WIDTH, HEIGHT = 540, 720

class EnhancedRacing:
    def __init__(self):
        self.root = tk.Tk()
        self.root.title("Enhanced Python Need for Speed Style Racing")
        self.canvas = tk.Canvas(self.root, width=WIDTH, height=HEIGHT, bg="black")
        self.canvas.pack()

        self.car_x = 245
        self.car_y = 570
        self.car_w = 52
        self.car_h = 92
        self.speed = 7
        self.road_offset = 0
        self.score = 0
        self.game_over = False
        self.keys = set()
        self.traffic = [
            [130, -100, 52, 92, "#d22828"],
            [245, -380, 52, 92, "#f0a01e"],
            [360, -650, 52, 92, "#50c8c8"]
        ]

        self.root.bind("<KeyPress>", self.press)
        self.root.bind("<KeyRelease>", self.release)

        self.loop()
        self.root.mainloop()

    def press(self, e):
        self.keys.add(e.keysym.lower())
        if e.keysym == "space" and self.game_over:
            self.restart()

    def release(self, e):
        self.keys.discard(e.keysym.lower())

    def draw_car(self, x, y, color, player=False):
        self.canvas.create_oval(x-5, y+self.car_h-8, x+self.car_w+5, y+self.car_h+8, fill="#000000", outline="")
        self.canvas.create_rectangle(x, y+10, x+self.car_w, y+self.car_h-8, fill=color, outline="")
        self.canvas.create_rectangle(x+5, y, x+self.car_w-5, y+self.car_h, fill=color, outline="white")
        self.canvas.create_rectangle(x+10, y+12, x+self.car_w-10, y+30, fill="#82d2ff", outline="")
        self.canvas.create_rectangle(x+10, y+52, x+self.car_w-10, y+70, fill="#82d2ff", outline="")
        for wx, wy in [(x-5,y+18),(x+self.car_w-5,y+18),(x-5,y+58),(x+self.car_w-5,y+58)]:
            self.canvas.create_oval(wx, wy, wx+10, wy+22, fill="black", outline="")
        if player:
            self.canvas.create_oval(x+15, y+self.car_h-3, x+23, y+self.car_h+17, fill="#ff7800", outline="")
            self.canvas.create_oval(x+30, y+self.car_h-3, x+38, y+self.car_h+17, fill="#ff7800", outline="")

    def update(self):
        if self.game_over:
            return

        if "left" in self.keys:
            self.car_x -= 6
        if "right" in self.keys:
            self.car_x += 6
        if "up" in self.keys and self.speed < 15:
            self.speed += 1
        if "down" in self.keys and self.speed > 4:
            self.speed -= 1

        self.car_x = max(88, min(400, self.car_x))
        self.road_offset = (self.road_offset + self.speed) % 130

        for car in self.traffic:
            car[1] += self.speed
            if car[1] > HEIGHT:
                car[1] = -random.randint(120, 720)
                car[0] = random.choice([130, 245, 360])

        self.check_collisions()
        self.score += 1

    def intersects(self, a, b):
        ax, ay, aw, ah = a
        bx, by, bw, bh = b
        return ax < bx+bw and ax+aw > bx and ay < by+bh and ay+ah > by

    def check_collisions(self):
        player = [self.car_x+5, self.car_y+5, self.car_w-10, self.car_h-10]
        for car in self.traffic:
            if self.intersects(player, [car[0]+5, car[1]+5, car[2]-10, car[3]-10]):
                self.game_over = True

    def restart(self):
        self.car_x = 245
        self.speed = 7
        self.score = 0
        self.game_over = False
        self.traffic = [
            [130, -100, 52, 92, "#d22828"],
            [245, -380, 52, 92, "#f0a01e"],
            [360, -650, 52, 92, "#50c8c8"]
        ]

    def draw(self):
        self.canvas.delete("all")
        self.canvas.create_rectangle(0, 0, WIDTH, HEIGHT, fill="#145523", outline="")
        self.canvas.create_rectangle(70, 0, 470, HEIGHT, fill="#232323", outline="")
        for y in range(-60 + self.road_offset, HEIGHT, 80):
            self.canvas.create_rectangle(70, y, 470, y+30, fill="#464646", outline="")
        self.canvas.create_rectangle(70, 0, 78, HEIGHT, fill="white", outline="")
        self.canvas.create_rectangle(462, 0, 470, HEIGHT, fill="white", outline="")

        for y in range(-80 + self.road_offset, HEIGHT, 130):
            self.canvas.create_rectangle(266, y, 274, y+75, fill="yellow", outline="")

        for i in range(14):
            x = 20 if i % 2 == 0 else 485
            y = (i * 75 + self.road_offset) % HEIGHT
            self.canvas.create_oval(x, y, x+34, y+26, fill="#0f3c19", outline="")
            self.canvas.create_rectangle(x+14, y+20, x+20, y+42, fill="#5a3719", outline="")

        for car in self.traffic:
            self.draw_car(car[0], car[1], car[4], False)

        self.draw_car(self.car_x, self.car_y, "#2355e6", True)

        self.canvas.create_rectangle(12, 12, 528, 54, fill="#000000", outline="#444444")
        self.canvas.create_text(85, 36, text=f"Score: {self.score}", fill="white", font=("Arial", 17, "bold"))
        self.canvas.create_text(270, 36, text="Arrows to drive", fill="white", font=("Arial", 15, "bold"))
        self.canvas.create_text(455, 36, text=f"Speed: {self.speed}", fill="white", font=("Arial", 17, "bold"))

        if self.game_over:
            self.canvas.create_rectangle(0, 295, WIDTH, 405, fill="#000000")
            self.canvas.create_text(WIDTH//2, 350, text="CRASH!", fill="red", font=("Arial", 42, "bold"))
            self.canvas.create_text(WIDTH//2, 385, text="Press SPACE to restart", fill="white", font=("Arial", 18, "bold"))

    def loop(self):
        self.update()
        self.draw()
        self.root.after(16, self.loop)

EnhancedRacing()

File: Racing/v2/Racing.java
import java.awt.*;
import java.awt.event.*;
import java.util.Random;
import javax.swing.*;

public class Racing extends JFrame {
    public Racing() {
        add(new RacingPanel());
        setTitle("Need for Speed Style Racing");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setResizable(false);
        pack();
        setLocationRelativeTo(null);
        setVisible(true);
    }

    public static void main(String[] args) {
        new Racing();
    }
}

class RacingPanel extends JPanel implements ActionListener, KeyListener {
    private final int WIDTH = 540, HEIGHT = 720;
    private Timer timer = new Timer(16, this);
    private int carX = 245, carY = 570, carW = 52, carH = 92;
    private int speed = 7, roadOffset = 0, score = 0;
    private boolean left, right, up, down, gameOver = false;
    private Random rand = new Random();
    private Rectangle[] traffic = {
        new Rectangle(130, -100, 52, 92),
        new Rectangle(245, -380, 52, 92),
        new Rectangle(360, -650, 52, 92)
    };

    public RacingPanel() {
        setPreferredSize(new Dimension(WIDTH, HEIGHT));
        setFocusable(true);
        addKeyListener(this);
        timer.start();
    }

    protected void paintComponent(Graphics g) {
        Graphics2D g2 = (Graphics2D)g;
        g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);

        drawScene(g2);
        drawTraffic(g2);
        drawCar(g2, carX, carY, new Color(35, 85, 230), true);
        drawHUD(g2);

        if (gameOver) {
            g2.setColor(new Color(0,0,0,180));
            g2.fillRect(0, 295, WIDTH, 110);
            g2.setColor(Color.RED);
            g2.setFont(new Font("Arial", Font.BOLD, 44));
            g2.drawString("CRASH!", 190, 350);
            g2.setColor(Color.WHITE);
            g2.setFont(new Font("Arial", Font.BOLD, 20));
            g2.drawString("Press SPACE to restart", 165, 385);
        }
    }

    private void drawScene(Graphics2D g2) {
        g2.setColor(new Color(20, 85, 35));
        g2.fillRect(0, 0, WIDTH, HEIGHT);

        g2.setColor(new Color(35, 35, 35));
        g2.fillRect(70, 0, 400, HEIGHT);

        g2.setColor(new Color(70,70,70));
        for (int y = -60 + roadOffset; y < HEIGHT; y += 80) {
            g2.fillRect(70, y, 400, 30);
        }

        g2.setColor(Color.WHITE);
        g2.fillRect(70, 0, 8, HEIGHT);
        g2.fillRect(462, 0, 8, HEIGHT);

        g2.setColor(Color.YELLOW);
        for (int y = -80 + roadOffset; y < HEIGHT; y += 130) {
            g2.fillRoundRect(266, y, 8, 75, 6, 6);
        }

        g2.setColor(new Color(15, 60, 25));
        for (int i = 0; i < 14; i++) {
            int x = i % 2 == 0 ? 20 : 485;
            int y = (i * 75 + roadOffset) % HEIGHT;
            g2.fillOval(x, y, 34, 26);
            g2.setColor(new Color(90,55,25));
            g2.fillRect(x+14, y+20, 6, 22);
            g2.setColor(new Color(15, 60, 25));
        }
    }

    private void drawTraffic(Graphics2D g2) {
        Color[] colors = {new Color(210,40,40), new Color(240,160,30), new Color(80,200,200)};
        for (int i = 0; i < traffic.length; i++) {
            Rectangle r = traffic[i];
            drawCar(g2, r.x, r.y, colors[i % colors.length], false);
        }
    }

    private void drawCar(Graphics2D g2, int x, int y, Color color, boolean player) {
        g2.setColor(new Color(0,0,0,90));
        g2.fillOval(x-5, y+carH-8, carW+10, 16);

        g2.setColor(color.darker());
        g2.fillRoundRect(x, y+10, carW, carH-18, 16, 16);
        g2.setColor(color);
        g2.fillRoundRect(x+5, y, carW-10, carH, 18, 18);

        g2.setColor(new Color(130,210,255));
        g2.fillRoundRect(x+10, y+12, carW-20, 18, 10, 10);
        g2.fillRoundRect(x+10, y+52, carW-20, 18, 10, 10);

        g2.setColor(Color.BLACK);
        g2.fillOval(x-5, y+18, 10, 22);
        g2.fillOval(x+carW-5, y+18, 10, 22);
        g2.fillOval(x-5, y+58, 10, 22);
        g2.fillOval(x+carW-5, y+58, 10, 22);

        if (player) {
            g2.setColor(new Color(255,120,0,180));
            g2.fillOval(x+15, y+carH-3, 8, 20);
            g2.fillOval(x+30, y+carH-3, 8, 20);
        }
    }

    private void drawHUD(Graphics2D g2) {
        g2.setColor(new Color(0,0,0,160));
        g2.fillRoundRect(12, 12, 516, 42, 15, 15);
        g2.setColor(Color.WHITE);
        g2.setFont(new Font("Arial", Font.BOLD, 19));
        g2.drawString("Score: " + score, 25, 39);
        g2.drawString("Speed: " + speed, 395, 39);
        g2.drawString("Arrows to drive", 190, 39);
    }

    public void actionPerformed(ActionEvent e) {
        if (!gameOver) {
            updateCar();
            updateRoad();
            updateTraffic();
            checkCollisions();
            score++;
        }
        repaint();
    }

    private void updateCar() {
        if (left) carX -= 6;
        if (right) carX += 6;
        if (up && speed < 15) speed++;
        if (down && speed > 4) speed--;
        if (carX < 88) carX = 88;
        if (carX > 400) carX = 400;
    }

    private void updateRoad() {
        roadOffset += speed;
        if (roadOffset >= 130) roadOffset = 0;
    }

    private void updateTraffic() {
        int[] lanes = {130, 245, 360};
        for (Rectangle r : traffic) {
            r.y += speed;
            if (r.y > HEIGHT) {
                r.y = -rand.nextInt(600) - 120;
                r.x = lanes[rand.nextInt(lanes.length)];
            }
        }
    }

    private void checkCollisions() {
        Rectangle player = new Rectangle(carX+5, carY+5, carW-10, carH-10);
        for (Rectangle r : traffic) {
            Rectangle hit = new Rectangle(r.x+5, r.y+5, r.width-10, r.height-10);
            if (player.intersects(hit)) gameOver = true;
        }
    }

    private void restart() {
        carX = 245; speed = 7; score = 0; gameOver = false;
        traffic[0].setLocation(130, -100);
        traffic[1].setLocation(245, -380);
        traffic[2].setLocation(360, -650);
    }

    public void keyPressed(KeyEvent e) {
        if (e.getKeyCode() == KeyEvent.VK_LEFT) left = true;
        if (e.getKeyCode() == KeyEvent.VK_RIGHT) right = true;
        if (e.getKeyCode() == KeyEvent.VK_UP) up = true;
        if (e.getKeyCode() == KeyEvent.VK_DOWN) down = true;
        if (e.getKeyCode() == KeyEvent.VK_SPACE && gameOver) restart();
    }

    public void keyReleased(KeyEvent e) {
        if (e.getKeyCode() == KeyEvent.VK_LEFT) left = false;
        if (e.getKeyCode() == KeyEvent.VK_RIGHT) right = false;
        if (e.getKeyCode() == KeyEvent.VK_UP) up = false;
        if (e.getKeyCode() == KeyEvent.VK_DOWN) down = false;
    }

    public void keyTyped(KeyEvent e) {}
}
